libsel4vm/src/guest_vspace.c

```diff
@@ -43,6 +43,14 @@ static int guest_vspace_map(vspace_t *vspace, seL4_CPtr cap, void *vaddr, seL4_C
     }
 
 #if defined(CONFIG_TK1_SMMU) || defined(CONFIG_IOMMU)
+    /* If an object doesn't have Read or Write access, there isn't a reason to put
+     * it in DMA space
+     */
+    if (rights.words[0] == seL4_NoRights.words[0]) {
```

Comment:
Is this solving some issue that was encountered? The same argument could be made to creating mappings without rights to the VMM memspace too, but its something that we don't restrict because it doesn't typically create a problem, and maybe the VMM could be trying to implement on-demand paging or something.

Reply:
I'm not sure... 

This was part of our backlog that we're trying to work through. If I remember correctly, we encountered some issue, but that was with the old SMMU driver that utilized iospaces, so this may not even be necessary. I figured i'd at least push this up for review and discussion, but we can always mark it as not needed.

Reply:
If only on-demand DMA mapping would work...oh well!

I'll close this PR.